packages/@aws-cdk/aws-ecs-patterns/lib/ecs/application-load-balanced-ecs-service.ts

```diff
@@ -70,27 +80,42 @@ export class ApplicationLoadBalancedEc2Service extends ApplicationLoadBalancedSe
   /**
    * Constructs a new instance of the ApplicationLoadBalancedEc2Service class.
    */
-  constructor(scope: Construct, id: string, props: ApplicationLoadBalancedEc2ServiceProps) {
+  constructor(scope: Construct, id: string, props: ApplicationLoadBalancedEc2ServiceProps = {}) {
     super(scope, id, props);
 
-    this.taskDefinition = new Ec2TaskDefinition(this, 'TaskDef', {
-      executionRole: props.executionRole,
-      taskRole: props.taskRole
-    });
+    if (props.taskDefinition && props.taskImageOptions) {
+      throw new Error('You must specify either a taskDefinition or taskImageOptions, not both.');
+    } else if (props.taskDefinition) {
+      this.taskDefinition = props.taskDefinition;
+    } else if (props.taskImageOptions) {
+      const taskImageOptions = props.taskImageOptions;
+      this.taskDefinition = new Ec2TaskDefinition(this, 'TaskDef', {
+        executionRole: taskImageOptions.executionRole,
+        taskRole: taskImageOptions.taskRole
+      });
```

Comment:
just noting that this seems repetitive but i guess with the separation of the *LBECSService there's not really a good way to dry this up...

Reply:
I thought about this earlier, we could create a common baseloadbalanced*Service construct, but I think we'd be pigeonholing ourselves again with what we were trying to avoid when we split the application and network load balanced constructs